.github/workflows/test.yml

```
@@ -151,237 +157,43 @@ jobs:
-          --quiet \
-          --ssh-flag="-o ServerAliveInterval=5" \
-          --command \
-          "docker run -t --name sync-past-checkpoint \
-          --mount type=bind,source=/mnt/disks/gce-containers-mounts/gce-persistent-disks/${{ env.CACHED_DISK_NAME }}-${{ env.GITHUB_SHA_SHORT }},target=/zebrad-cache \
-          ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} \
-          cargo test --locked --release --features enable-sentry,test_sync_past_mandatory_checkpoint_${{ env.NETWORK }} --manifest-path zebrad/Cargo.toml sync_past_mandatory_checkpoint_${{ env.NETWORK }}"
-
-          EXIT_CODE=$(\
-          gcloud compute ssh \
-          sync-past-checkpoint-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
-          --zone ${{ env.ZONE }} \
-          --quiet \
-          --ssh-flag="-o ServerAliveInterval=5" \
-          --command="docker wait sync-past-checkpoint")
-
-          exit ${EXIT_CODE}
-
-      - name: Delete test instance
-        # We don't want to leave a failed instance in GCP using resources
-        if: always()
-        continue-on-error: true
-        run: |
-          INSTANCE=$(gcloud compute instances list --filter=sync-past-checkpoint-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} --format='value(NAME)')
-          if [ -z "${INSTANCE}" ]; then
-            echo "No instance to delete"
-          else
-            gcloud compute instances delete "${INSTANCE}" --zone "${{ env.ZONE }}" --delete-disks all --quiet
-          fi
+    if: ${{ github.event.inputs.regenerate-disks != 'true' }}
+    needs: build
+    uses: ./.github/workflows/gcp-test-deploy.yml
+    with:
+      test_id: sync-past-checkpoint
+      test_description: Test full validation sync from a cached state
+      test_variables: '-e TEST_CHECKPOINT_SYNC=1 -e ZEBRA_FORCE_USE_COLOR=1'
+      disk_suffix: checkpoint
+      saves_to_disk: false
+      needs_constants: false
+
+  # Test that Zebra can run a full mainnet sync after a PR is approved
+  test-full-sync:
+    # TODO change `github.ref_name == 'main'` to `startsWith(github.head_ref, 'mergify/merge-queue/')`
+    # to only run on Mergify head branches, and on manual dispatch:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-based-on-the-head-or-base-branch-of-a-pull-request-1
+    if: ${{ (github.event_name == 'push' && github.ref_name == 'main') || github.event.inputs.run-full-sync == 'true' }}
+    needs: build
+    uses: ./.github/workflows/gcp-test-deploy.yml
+    with:
+      test_id: full-sync-to-tip
+      test_description: Test a full sync up to the tip
+      test_variables: '-e TEST_FULL_SYNC=1 -e ZEBRA_FORCE_USE_COLOR=1 -e FULL_SYNC_MAINNET_TIMEOUT_MINUTES=600'
+      disk_suffix: tip
+      saves_to_disk: true
+      needs_constants: false
+      height_grep_text: 'finished initial sync to chain tip, using gossiped blocks sync_percent=100.000 % current_height=Height'
```

I wouldn't want to rely on the exact format here:
```suggestion
      height_grep_text: 'finished initial sync to chain tip, using gossiped blocks sync_percent=100.* current_height=Height'
```